File: ModernizacionPersonas.Web/ClientApp/src/app/models/tomador.ts
export class Tomador {
  codigoCotizacion: number;
  codigoTomador: number;
  codigoTipoDocumento: number;
  numeroDocumento: string;
  nombres: string;
  primerApellido: string;
  segundoApellido: string;
  direccion: string;
  codigoPais: number;
  codigoDepartamento: number;
  codigoMunicipio: number;
  email: string;
  nombreContacto: string;
  telefono1Contacto: string;
  telefono2Contacto: string;
  codigoActividad: string;
  licitacion: boolean;
  aseguradoraActual: string;
  descripcionTipoRiesgo: string;

  static copy(model: Tomador): Tomador {
    let t = new Tomador();
    return Object.assign(t, model)
  }

  isEqual(this: Tomador, target: Tomador) {
    return this.codigoTipoDocumento === target.codigoTipoDocumento
      && this.primerApellido === target.primerApellido
      && this.numeroDocumento === target.numeroDocumento
      && false;
  }
}

File: ModernizacionPersonas.Entities/Tomador.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ModernizacionPersonas.Entities
{
    public class Tomador
    {
        public int CodigoCotizacion { get; set; }
        public int CodigoTomador { get; set; }
        public int CodigoTipoDocumento { get; set; }
        public string NumeroDocumento { get; set; }
        public string Nombres { get; set; }
        public string PrimerApellido { get; set; }
        public string SegundoApellido { get; set; }
        public string Direccion { get; set; }
        public int CodigoPais { get; set; }
        public int CodigoDepartamento { get; set; }
        public int CodigoMunicipio { get; set; }
        public string Email { get; set; }
        public string NombreContacto { get; set; }
        public string Telefono1Contacto { get; set; }
        public string Telefono2Contacto { get; set; }
        public int CodigoActividad { get; set; }
        public string Actividad { get; set; }
        public bool Licitacion { get; set; }
        public string AseguradoraActual { get; set; }
        public string DescripcionTipoRiesgo { get; set; }
        public string TomadorSlip { get; set; }

        public override int GetHashCode()
        {
            var hash = 0;
            hash = (hash * 397) ^ NumeroDocumento.GetHashCode();

            return hash;
        }

        public override bool Equals(object obj)
        {
            Tomador tomador = obj as Tomador;
            if (tomador.CodigoTomador == 0) return false;

            var isEqual = this.GetHashCode() == obj.GetHashCode();
            return false;
        }
    }
}
